### DISCUSSION TITLE
Interactive mode

### DISCUSSION THREAD

**Discussion by denilsonsa** [Kalanyr/gogrepo#102] [Ideas]
While I do like the pure command-line interface and I think it should always exist, I wonder if some minor interactivity would make the tool easier to use.

For instance, I wouldn't mind having a parameter to let me toggle which "extras" types I would like to download (see #101). It could be [easily achieved](https://python-inquirer.readthedocs.io/en/latest/examples.html#checkbox-py) using [python-inquirer](https://github.com/magmax/python-inquirer).

Alternatively, [textual](https://github.com/Textualize/textual/) seems to be a very complete library/framework for text-based user interfaces, but that will probably require a lot more code than just a few lines of the `inquirer`.

---

Footnote: After I finish downloading what I want, I'll probably not use this tool again for quite a long time. So, don't feel pressured to build it for me; I'm just sharing the idea because I thought it would be cool and useful (and possibly simple enough to implement).

**Comment by Kalanyr** [comment]
I do have a GUI version that I put aside for reasons I no longer remember (
probably some sort of cross-version or cross-platform comparability issue
). I should probably upload that for anyone who wants to mess with it.

Inquirer looks minimal enough to put on my  to do list.

**Comment by denilsonsa** [comment]
I used inquirer once, it was awesomely simple and solved my need perfectly in that moment.

I also like using [tqdm](https://github.com/tqdm/tqdm) for (multiple) progress bars (and this one even works on Jupyter notebooks). Not that you have to use it, but it's a nice little library.
